admin_app/src/app/content/edit/page.tsx

```diff
@@ -268,8 +276,54 @@ const ContentBox = ({
       setSnackMessage(`Tag "${tag}" already exists`);
     }
   };
+  const openDeleteConfirmModal = (tag: Tag) => {
+    setTagToDelete(tag);
+    setOpenDeleteModal(true);
+  };
+  const handleDeleteTag = () => {
+    if (tagToDelete) {
+      deleteTag(tagToDelete.tag_id).then(() => {
+        setRefreshKey((prevKey) => prevKey + 1);
+        setOpenDeleteModal(false);
+        handleTagsChange(
+          contentTags.filter((tag) => tag.tag_id !== tagToDelete.tag_id),
+        );
+        setTagToDelete(null);
+      });
+    }
+  };
   return (
     <Layout.FlexBox>
+      <Dialog
+        open={openDeleteModal}
+        onClose={() => setOpenDeleteModal(false)}
+        aria-labelledby="confirm-delete-dialog-title"
+        aria-describedby="confirm-delete-dialog-description"
+      >
+        <DialogTitle id="confirm-delete-dialog-title">
+          Are you sure you want to delete tag {tagToDelete?.tag_name}?
+        </DialogTitle>
+        <DialogContent>
+          <DialogContentText id="confirm-delete-dialog-description">
+            This tag will be removed from all contents and will no longer be an
+            option. This action cannot be undone.
+          </DialogContentText>
+        </DialogContent>
+        <DialogActions sx={{ marginBottom: 1, marginRight: 1 }}>
+          <Button onClick={() => setOpenDeleteModal(false)} color="primary">
+            Cancel
+          </Button>
+          <Button
+            onClick={() => handleDeleteTag()}
+            autoFocus
+            variant="contained"
+            color="error"
+            startIcon={<Delete />}
+          >
+            Delete
+          </Button>
+        </DialogActions>
+      </Dialog>
```

Comment:
I committed some aesthetic changes to make it match content delete modal (easier than leaving review comments).

Also, could you pull this out and have it as a separate component? This return statement is already quite big.